src/main/scala/com/fulcrumgenomics/umi/ReviewConsensusVariants.scala

```diff
@@ -25,22 +25,99 @@
 package com.fulcrumgenomics.umi
 
 import java.nio.file.Path
+import java.util.Collections
 
 import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.bam.BaseCounts
 import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
-import com.fulcrumgenomics.util.Io
+import com.fulcrumgenomics.umi.ReviewConsensusVariants._
+import com.fulcrumgenomics.util.{Io, Metric}
 import dagr.commons.util.LazyLogging
+import dagr.sopt
 import dagr.sopt.{arg, clp}
+import htsjdk.samtools.SamPairUtil.PairOrientation
+import htsjdk.samtools._
 import htsjdk.samtools.reference.{ReferenceSequenceFile, ReferenceSequenceFileFactory}
 import htsjdk.samtools.util.IOUtil.{VCF_EXTENSIONS => VcfExtensions}
-import htsjdk.samtools.util.{IntervalList, SequenceUtil}
-import htsjdk.samtools.{QueryInterval, SAMFileWriterFactory, SAMRecord, SamReaderFactory}
+import htsjdk.samtools.util._
+import htsjdk.variant.variantcontext.VariantContext
 import htsjdk.variant.vcf.VCFFileReader
 
+import scala.collection.JavaConverters._
 import scala.collection.mutable
 import scala.collection.mutable.ListBuffer
 
-@clp(group=ClpGroups.Umi, description=
+object ReviewConsensusVariants {
+
+  /**
+    * Metric class that is used to output detailed information on each consensus read that carries a variant allele
+    * @param chrom the chromosome on which the variant exists
+    * @param pos the position of the variant
+    * @param ref the reference allele at the position
+    * @param genotype the genotype of the sample in question
+    * @param filters the set of filters applied to the variant in the VCF
+    * @param A the count of A observations at the variant locus
+    * @param C the count of C observations at the variant locus
+    * @param G the count of G observations at the variant locus
+    * @param T the count of T observations at the variant locus
+    * @param N the count of N observations at the variant locus
+    * @param consensus_read the consensus read name for which the following fields contain values
+    * @param consensus_insert a description of the insert that generated the consensus read
+    * @param consensus_call the base call from the consensus read
+    * @param consensus_qual the quality score from the consensus read
+    * @param a the number of As in raw-reads contributing to the consensus base call at the variant site
+    * @param c the number of Cs in raw-reads contributing to the consensus base call at the variant site
+    * @param g the number of Gs in raw-reads contributing to the consensus base call at the variant site
+    * @param t the number of Ts in raw-reads contributing to the consensus base call at the variant site
+    * @param n the number of Ns in raw-reads contributing to the consensus base call at the variant site
+    */
+  case class ConsensusVariantReviewInfo
+  ( // First set is all the same for each variant
+    chrom: String, pos: Int, ref: String, genotype: String, filters: String, A: Int, C: Int, G: Int, T: Int, N: Int,
```

Comment:
It would be nice, but since it also has to play nice with how Metric works, I'm afraid of losing a lot of time trying to find a solution that "just works".